### DISCUSSION TITLE
Rendering Disclosure as a Fragment

### DISCUSSION THREAD

**Discussion by cubecleveland** [ariakit/ariakit#924] [Questions] [answered]
HI I have the folloiiung code
<DialogDisclosure {...dialog}>
        {(props) => (
          <Button
            onClick={() =>
              mutateRestore.mutate(ac.id, {
                onSuccess: () => {
                  setEvents((prev) => [...prev, "DONE"]);
                },
              })
            }
          >
            <ArrowClockwise />
            Restore
          </Button>
        )}
      </DialogDisclosure>

I am not using Reakit Button but the button is my own.
Under the current code the button is wrapped in a div... id like to actually dissolve the doiv and sinmply render the button i assume u need to do it when passing as={Fragment} but the console is giving errors.
CAn u put an example of how to render the disclosure as a Fragment ?

**Comment by diegohaz** [comment, marked as answer]
You can't do `<DialogDisclosure as={Fragment} />` because `Fragment` doesn't accept HTML props and doesn't render an actual HTML element.

In your case, you can do `<DialogDisclosure as={Button} onClick={() => mutate...} />`, `DialogDisclosure` will render your `Button` and will pass the props down. You have to make sure your custom `Button` component forwards the `ref` prop and accepts all HTML props.
